Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -402,6 +482,20 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
             }
             return lib;
         }
+    
+        bool operator==(const Pose& lhs, const Pose& rhs) {
+            return abs(lhs.Position.X - rhs.Position.X) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Position.Y - rhs.Position.Y) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Position.Z - rhs.Position.Z) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Orientation.X - rhs.Orientation.X) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Orientation.Y - rhs.Orientation.Y) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Orientation.Z - rhs.Orientation.Z) < FLOAT_COMPARISON_THRESHOLD
+                   && abs(lhs.Orientation.W - rhs.Orientation.W) < FLOAT_COMPARISON_THRESHOLD;
+        }
+
+        bool operator!=(const Pose& lhs, const Pose& rhs) {
+            return !(lhs == rhs);
+        }
```

Comment:
Is this the same as the one for ARCore? Should we put this in a shared header (not xr.h, but a shared implementation header) that both files can use?

Reply:
Added a new XRHelpers.h which implements this, and removed the individual implementations.